libnopegl/src/glsl/helper_noise.glsl

```diff
@@ -85,6 +85,20 @@ float random_grad(uvec2 x, vec2 f)
  */
 float random_grad(uvec3 x, vec3 f)
 {
+    /*
+    Adapted from https://web.archive.org/web/20180616011332/http://www.scratchapixel.com/lessons/procedural-generation-virtual-worlds%20/perlin-noise-part-2
+
+    float value = u32tof32(hash(x));
+    float theta = acos(2.0*value - 1.0);
+    float phi = 2.0 * value * 3.14;
+    vec3 grad = vec3(
+        cos(phi) * sin(theta),
+        sin(phi) * sin(theta),
+        cos(theta)
+    );
+
+    return dot(grad, f);
```

Comment:
It seems the value must be different for theta and phi, so maybe:
```glsl
    float r0 = u32tof32(0x00010001 * (hash(x)       & 0xffff));
    float r1 = u32tof32(0x00010001 * (hash(x) >> 16 & 0xffff));
    float theta = acos(2.0*r0 - 1.0);
    float phi = 2.0 * ngli_pi * r1;
    vec3 grad = vec3(cos(phi) * sin(theta), sin(phi) * sin(theta), cos(theta));
    return dot(grad, f);
```

Reply:
Fixed.